org.adempiere.base/src/org/compiere/model/MOrder.java

```diff
@@ -2388,7 +2388,7 @@ protected MOrder createCounterDoc()
 		{
 			MOrderLine counterLine = counterLines[i];
 			counterLine.setOrder(counter);	//	copies header values (BP, etc.)
-			counterLine.setPrice();
+			counterLine.setPrice(counterLine.getPriceEntered());
```

Comment:
This is unnecessary (if priceentered have been copied over from source line, priceactual should have been copied over as well). If the concern is not all price related value have been copied over from source line (perhaps due to changes to ad_column's allow copy flag), changes should be added to copyLinesFrom instead, i.e if counter==true, make sure all price related value from source line have been copied over to counter line.

Reply:
I was a proposal of Nicolas because you can have "allow copy" set to NO .

Reply:
@nmicoud    discussion

Reply:
yeah but as I mention above, that's insufficient for the "allow copy" set to no concern since you can have that set to no for price entered as well.

Reply:
hello
nb: I did that change for a particular customer - not search for a nice/generic solution.
I would prefer hengsin's approach

Reply:
IMO is simply just to drop that line - the price is already set above.